### ISSUE TITLE
After update to Python 3.11, I get ugly warning:  Could not find platform dependent libraries <exec_prefix>

### ISSUE THREAD

**Issue by zewpo** [vim/vim-win32-installer#293]
### Steps to reproduce

1.  start vim with `vim --clean`

![image](https://user-images.githubusercontent.com/10286211/202350456-aee4db95-cb2a-4f90-82f7-1cecd93dd30e.png)

2.  run a python3 command, eg   `:python3 print("hello world")`
![image](https://user-images.githubusercontent.com/10286211/202350563-cd10be1c-d446-4e08-8454-a9891e10c735.png)

3.  observe error:  
```
Could not find platform dependent libraries <exec_prefix>
hello world
```
![image](https://user-images.githubusercontent.com/10286211/202350655-1760c722-3dea-4c75-b87c-1f1d17d7e766.png)

Strange thing is, despite the warning message, it seems to work anyway as it does actually print "hello world"

### Expected behaviour

Expecting not to see any warning at all about `Could not find platform dependent libraries <exec_prefix>`

### Version of Vim

commit [Update interface for Python 3 (3.10 to 3.11)] #8fe4cf342cfcb0a7d5902b3cbc99a6b494ea32c5

### Environment

Reference
https://github.com/vim/vim-win32-installer/commit/8fe4cf342cfcb0a7d5902b3cbc99a6b494ea32c5

Note, I also get this problem when I compile vim against 3.11 in my own dev environment.

Now, another strange thing is that if I compile Vim against 3.10, and then use 3.11 in my path at runtime, it all works fine without any error.   It seems that the error has something to do with compiling Vim against  3.11.   HANG ON.  EDIT - sorry, scratch that - it was working because I had python310.dll copied into the vim runtime.   So, NO, it looks like if it is compiled with python310, then it needs python310 at runtime.

**Comment by chrisbra** [comment]
hm, not sure. According to google, you may need to set `$PYTHONHOME`

**Comment by zewpo** [comment]
@chrisbra  Can you confirm if you tried to replicate the issue on your end?  I've tried it on my PC as well as a clean install of windows and that had only one python installation (3.11) - and I still got the same problem.

**Comment by zewpo** [comment]
Solved after upgrading to Python 3.11.1
